ionex2kml/src/cli.rs

```diff
@@ -0,0 +1,119 @@
+use clap::{
+    Arg, //ArgAction,
+    ArgMatches,
+    ColorChoice,
+    Command,
+};
+use log::error;
+use std::collections::HashMap;
+use std::str::FromStr;
+use thiserror::Error;
+
+use kml::types::KmlVersion;
+
+#[derive(Debug, Error)]
+pub enum CliError {
+    #[error("file type \"{0}\" is not supported")]
+    FileTypeError(String),
+    #[error("failed to parse ionex file")]
+    IonexError(#[from] rinex::Error),
+    #[error("failed to generate kml content")]
+    GpxError(#[from] kml::Error),
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Cli {
+    matches: ArgMatches,
+}
+
+impl Cli {
+    pub fn new() -> Self {
+        Self {
+            matches: {
+                Command::new("ionex2kml")
+                    .author("Guillaume W. Bres, <guillaume.bressaix@gmail.com>")
```

Comment:
https://docs.rs/clap/latest/clap/macro.command.html or https://docs.rs/clap/latest/clap/macro.crate_authors.html and friends?